### PR TITLE
SRL fine-tuning and evaluation

### PR DESCRIPTION
**Pull request by rexylic**
`train.py` file handles pre-training and fine-tuning. Evaluation is done with the existing `pr_score` metric of `eval.py`. 

Please let me know if the refactoring is too much, I can come up with a version that is the closest to main. 

Thank you.

### CODE REVIEW

File: main_clone.py [hunk 1]
```diff
@@ -0,0 +1,366 @@
+import argparse
```
Comment (Aatlantise) [inline main_clone.py:1]: It looks like there's much overlap between this file and `models/t5.py` and `train.py.` We want to minimize duplicate code as much as possible bc that makes the codebase much easier to read and maintain.

File: train.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            eval_batch_size = eval_batch_size,
+            num_workers = num_workers,
+        ))
+        logger.info('Tested model')
+    except Exception as e:
+        # Print the error message but don't stop the program
+        logger.warning('Trainer test did not run. Error below:')
+        logger.warning(e)
+
+if __name__ == '__main__':
```
Comment (Aatlantise) [inline train.py:127]: If you're keeping `train.py`, could you parametrize the script so that PR training and downstream finetuning (e.g. SRL) can be controlled? You can use `argparse` to achieve this.

**Review by Aatlantise — CHANGES_REQUESTED** [review]
Overall, I think having a script for each data / model architecture is good. If you can clean up the overlap,  that would be great! Thanks. `main.py`, `main_clone.py`, and {`train.py`, `t5.py`} seem to be triply-redundant right now.
